main/tests/cypress/cypress/integration/project/grid/column/facet/facets.spec.js

```diff
@@ -1,184 +1,417 @@
+		// we force include/exclude to be visible
+		// This test focus solely on ensuring that filters are effectively applied to the grid
+		cy.loadAndVisitProject('food.small')
+		cy.columnActionClick('Shrt_Desc', ['Facet', 'Text facet'])
+
+		// force visibility
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice-toggle')
+			.invoke('attr', 'style', 'visibility:visible')
+
+		// include ALLSPICE,GROUND, and check rows
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice')
+			.contains('ALLSPICE,GROUND')
+			.parent()
+			.find('.facet-choice-toggle')
+			.click()
+		cy.getCell(0, 'Shrt_Desc').should('to.contain', 'ALLSPICE,GROUND')
+		cy.get('#tool-panel').contains('1 matching rows')
+
+		// OR ir refreshing facets, need to show the toggle again
+		cy.waitForOrOperation()
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice-toggle')
+			.invoke('attr', 'style', 'visibility:visible')
+
+		// include CELERY SEED, and check rows
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice')
+			.contains('ANISE SEED')
+			.parent()
+			.find('.facet-choice-toggle')
+			.click()
+		cy.getCell(1, 'Shrt_Desc').should('to.contain', 'ANISE SEED')
+		cy.get('#tool-panel').contains('2 matching rows')
+
+		// OR ir refreshing facets, need to show the toggle again
+		cy.waitForOrOperation()
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice-toggle')
+			.invoke('attr', 'style', 'visibility:visible')
+
+		// include a third one, CELERY SEED, and check rows
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice')
+			.contains('BUTTER OIL,ANHYDROUS')
+			.parent()
+			.find('.facet-choice-toggle')
+			.click()
+		cy.getCell(0, 'Shrt_Desc').should('to.contain', 'BUTTER OIL,ANHYDROUS') // this row is added first
+		cy.get('#tool-panel').contains('3 matching rows')
+
+		// OR ir refreshing facets, need to show the toggle again
+		cy.waitForOrOperation()
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice-toggle')
+			.invoke('attr', 'style', 'visibility:visible')
+
+		// EXCLUDE ALLSPICE,GROUND
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice')
+			.contains('ALLSPICE,GROUND')
+			.parent()
+			.find('.facet-choice-toggle')
+			.click()
+		cy.get('#tool-panel').contains('2 matching rows')
+	})
+
+	it('Test include/exclude, invert', function () {
+		cy.loadAndVisitProject('food.small')
+		cy.columnActionClick('Shrt_Desc', ['Facet', 'Text facet'])
+
+		// do a basic facetting, expect 1 row
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice-toggle')
+			.invoke('attr', 'style', 'visibility:visible')
+		cy.getFacetContainer('Shrt_Desc')
+			.find('.facet-choice')
+			.contains('ALLSPICE,GROUND')
+			.parent()
+			.find('.facet-choice-toggle')
+			.click()
+		cy.getCell(0, 'Shrt_Desc').should('to.contain', 'ALLSPICE,GROUND')
+		cy.get('#tool-panel').contains('1 matching rows')
+
+		// now invert, expect 198
+		cy.getFacetContainer('Shrt_Desc').find('a[bind="invertButton"]').click()
```

Comment:
```suggestion
		cy.getFacetContainer('Shrt_Desc').get('a[bind="invertButton"]').as('invertButton')
		cy.get('@invertButton').click()
```

Hmm, save yourself lots of typing throughout tests by using more aliases `@` syntax, such as above.
find() gets descendents but you cannot alias it like with get()
Please read:
https://docs.cypress.io/api/commands/get.html#Alias
https://docs.cypress.io/guides/core-concepts/variables-and-aliases.html#Aliases

Reply:
I wonder if Cypress has some linting help to spot things like all those same find()'s and suggest to use alias?  Kinda like Sonarlint does with Java where there's sometimes repeated statements and then asking to refactor to simplify?  Maybe sometime wrote a plugin for Cypress to catch things like that, dunno.

Reply:
It might also make sense, or perhaps not, to use shorthand notation via this plugin: https://github.com/FlorianGoussin/cypress-cy-select  which I found while browsing through all the various plugins here: https://docs.cypress.io/plugins/

Reply:
Regarding use of `as`, yes it will work in that case (tested), but it's actually a double edge sword.
As soon as an element is detached from the DOM, Cypress will loose the reference, and we never know, and can't know when does OpenRefine attach/detach/replace entire portions of the dom.
So it would make the tests difficult to troubleshoot, especially for newcomers. 
I would rather keep this verbose syntax, for consistency across tests and scenarios.

Regarding adding `data-cy`, well, I am not sure it's a good idea or not. For now I've recommenced to never touch OpenRefine codebase, so it makes PRs easy to merge, and allows newcomers to write some tests without worrying to break anything.
What could be quite usefull actually, is to have a better syntax to select DOM elements by the `bind` attribute, because OR heavily rely on this to structure the DOM, not sure yet what's the best way to implement this

Reply:
> I wonder if Cypress has some linting help to spot things like all those same find()'s and suggest to use alias? Kinda like Sonarlint does with Java where there's sometimes repeated statements and then asking to refactor to simplify? Maybe sometime wrote a plugin for Cypress to catch things like that, dunno.

There issue #3536 where we can discuss linting

Reply:
OK thanks for explaining.
I thought I saw some plugins to help you select for `bind` attributes?